src/Exception/BadRequestHttpException.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/**
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Exception;
+
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException as BaseBadRequestHttpException;
+
+/**
+ * This exception is thrown when there is an issue with the request.
+ *
+ * @author Christian Schiffler <https://github.com/discordier>
+ */
+class BadRequestHttpException extends BaseBadRequestHttpException
```

Comment:
Exactly. If a third party bundle throws any exception you are responsible of catching and chaining/rethrowing it.
If any "unknown" exception is encountered in our "root exception handler", it is supposed to be a HTTP_INTERNAL_SERVER_ERROR as it is not intended for Contao and/or does not follow the normal workflow.